discovery-provider/src/tasks/index_spl_token.py

```diff
@@ -170,53 +173,87 @@ def parse_spl_token_transaction(
                 f"index_spl_token.py | {tx_sig} No transfer checked instruction found"
             )
             return None
+
         tx_message = cast(Transaction, result.transaction.transaction).message
-        instruction = get_valid_instruction(tx_message, meta, SPL_TOKEN_ID)
-        if not instruction:
-            logger.error(
-                f"index_spl_token.py | {tx_sig} No valid instruction for spl token program found"
-            )
-            return None
+        account_keys = list(map(lambda key: str(key), tx_message.account_keys))
+        if version == 0 and meta.loaded_addresses:
+            writable = meta.loaded_addresses.writable or []
+            readonly = meta.loaded_addresses.readonly or []
+            account_keys.extend(str(key) for key in writable + readonly)
```

Comment:
might be worth having a helper abstract the ordering away from the caller, eg accept a `loaded_addresses` arg in `get_solana_tx_token_balance_changes` or make separate helper `get_all_account_keys` or something. Just so we don't have to remember this ordering etc every indexer

Reply:
i like that